vm/package/execute.rs

```diff
@@ -0,0 +1,342 @@
+    pub fn new(transaction: Transaction<N>, address: Address<N>, program_id: ProgramID<N>) -> Result<Self> {
+        ensure!(
+            matches!(transaction, Transaction::Execute(_, _, _)),
+            "Cannot create an execution request with a deploy transaction"
+        );
+        Ok(Self { transaction, address, program_id })
+    }
+
+    /// Sends the request to the given endpoint.
+    pub fn send(&self, endpoint: &str) -> Result<ExecuteResponse<N>> {
+        Ok(ureq::post(endpoint).send_json(self)?.into_json()?)
+    }
+
+    /// Returns the deployment transaction.
+    pub const fn transaction(&self) -> &Transaction<N> {
+        &self.transaction
+    }
+
+    /// Returns the program address.
+    pub const fn address(&self) -> &Address<N> {
+        &self.address
+    }
+
+    /// Returns the imports.
+    pub const fn program_id(&self) -> &ProgramID<N> {
+        &self.program_id
+    }
+}
+
+impl<N: Network> Serialize for ExecutionRequest<N> {
+    /// Serializes the deploy request into string or bytes.
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut request = serializer.serialize_struct("ExecuteRequest", 3)?;
+        // Serialize the deployment.
+        request.serialize_field("transaction", &self.transaction)?;
+        // Serialize the address.
+        request.serialize_field("address", &self.address)?;
+        // Serialize the program id.
+        request.serialize_field("program_id", &self.program_id)?;
+        request.end()
+    }
+}
+
+impl<'de, N: Network> Deserialize<'de> for ExecutionRequest<N> {
+    /// Deserializes the deploy request from a string or bytes.
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        // Parse the request from a string into a value.
+        let request = serde_json::Value::deserialize(deserializer)?;
+        // Build the execution request.
+        let execution_request = Self::new(
+            // Retrieve the execution transaction.
+            serde_json::from_value(request["transaction"].clone()).map_err(de::Error::custom)?,
+            // Retrieve the address of the program.
+            serde_json::from_value(request["address"].clone()).map_err(de::Error::custom)?,
+            // Retrieve the program ID.
+            serde_json::from_value(request["program_id"].clone()).map_err(de::Error::custom)?,
+        );
+        match execution_request {
+            Ok(execution_request) => Ok(execution_request),
+            Err(error) => Err(de::Error::custom(error.to_string())),
+        }
+    }
+}
+
+pub struct ExecuteResponse<N: Network> {
+    transaction: Transaction<N>,
+}
+
+impl<N: Network> ExecuteResponse<N> {
+    /// Initializes a new deploy response.
+    pub fn new(transaction: Transaction<N>) -> Result<Self> {
+        ensure!(
+            matches!(transaction, Transaction::Execute(_, _, _)),
+            "Cannot create an execution response with a deploy transaction"
```

Comment:
```suggestion
            "Cannot create an execution response with a non-execution transaction"
```
^We intend to support other transaction types in the future, hence the greater specificity.